dlt/extract/hints.py

```diff
@@ -506,6 +513,22 @@ def _merge_merge_disposition_dict(dict_: Dict[str, Any]) -> None:
                 "row_key": False,
             }
 
+    @staticmethod
+    def _merge_incremental_column_hint(dict_: Dict[str, Any]) -> None:
+        incremental = dict_.pop("incremental")
+        if isinstance(incremental, Incremental):
+            incremental_hint = incremental.to_table_hint()
```

Comment:
the ticket asked to put the full incremental info into table now when I think of it - it will be confusing. it describes the behavior of the data source, not the data in the actual table. my take is to just set incremental to `True` if it is present on a column,

it is sufficient for what we plan upstream. we could add row order later if we really need it (using `x-` annotation)

I do not thing this impacts this PR too much